haystack-commons/src/main/java/com/expedia/www/haystack/metrics/HaystackGraphiteNamingConvention.java

```diff
@@ -0,0 +1,67 @@
+package com.expedia.www.haystack.metrics;
+
+import com.netflix.servo.Metric;
+import com.netflix.servo.annotations.DataSourceType;
+import com.netflix.servo.monitor.MonitorConfig;
+import com.netflix.servo.publish.graphite.GraphiteNamingConvention;
+import com.netflix.servo.tag.Tag;
+import com.netflix.servo.tag.TagList;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import static com.expedia.www.haystack.metrics.MetricObjects.TAG_KEY_CLASS;
+import static com.expedia.www.haystack.metrics.MetricObjects.TAG_KEY_SUBSYSTEM;
+
+public class HaystackGraphiteNamingConvention implements GraphiteNamingConvention {
+    static final String HOST_NAME_UNKNOWN_HOST_EXCEPTION = "HostName-UnknownHostException";
+    static final String MISSING_TAG = "MISSING_TAG_%s";
+    static final String MISSING_VALUE = "MISSING_VALUE_%s";
+    static final String METRIC_FORMAT = "%s.%s.%s.%s_%s";
+
+    static Factory factory = new Factory(); // will be mocked out in unit tests
+    static final String LOCAL_HOST_NAME = getLocalHost();
```

Comment:
Doing so requires the code instantiating this instance (MetricPublishing.Factory) to deal with the UnknownHostException. I'll play with this and see how it feels.

Reply:
I don't think that passing in localHostName makes things any better, because the static InetAddress.getLocalHost() throws UnknownHostException, and the desire to handle that elegantly (substituting a string documenting the problem where the host name should be) is better hidden deep, and since this same class is handling missing tags and values in a similar way, I think it belongs here.

Reply:
I think it will make HaystackGraphiteNamingConvention cleaner. Yes, we will still have InetAddress.getLocalHost and its exception handling - like you do now - but that will move into MetricPublishing.Factory  and keep this class clean and easy to instantiate/test with a constructor. Thoughts?

Reply:
I don't have a problem doing it that way and will make the change.

Reply:
Change made.